tap_square/client.py

```diff
@@ -39,43 +39,91 @@ def log_backoff(details):
     LOGGER.warning('Error receiving data from square. Sleeping %.1f seconds before trying again', details['wait'])
 
 
+def write_config(config, config_path, data):
+    """
+    Updates the provided filepath with json format of the `data` object
+    """
+    config.update(data)
+    with open(config_path, "w") as tap_config:
+        json.dump(config, tap_config, indent=2)
+    return config
+
+
+def require_new_access_token(access_token, client):
+    """
+    Checks if the access token needs to be refreshed
+    """
+    # If there is no access token, we need to generate a new one
+    if not access_token:
+        return True
+
+    authorization = f"Bearer {access_token}"
+
+    with singer.http_request_timer("Check access token expiry"):
+        response = client.o_auth.retrieve_token_status(authorization)
+
+    if response.is_error():
+        error_message = response.errors if response.errors else response.body
+        LOGGER.error(error_message)
+        return True
```

Comment:
On failure, why we are not throwing an exception instead of just logging the error?